records/state/state.go

```diff
@@ -123,12 +131,22 @@ var sources = map[string]func(*Task) []string{
 func hostIPs(t *Task) []string { return []string{t.SlaveIP} }
 
 // networkInfoIPs returns IP addresses from a given Task's
-// []Status.ContainerStatus.[]NetworkInfos.IPAddress
+// []Status.ContainerStatus.[]NetworkInfos.[]IPAddresses.IPAddress
 func networkInfoIPs(t *Task) []string {
 	return statusIPs(t.Statuses, func(s *Status) []string {
 		ips := make([]string, len(s.ContainerStatus.NetworkInfos))
-		for i, netinfo := range s.ContainerStatus.NetworkInfos {
-			ips[i] = netinfo.IPAddress
+		for _, netinfo := range s.ContainerStatus.NetworkInfos {
+			if len(netinfo.IPAddresses) > 0 {
```

Comment:
Hm, just re-reviewed the mesos-0.26-rc1 code:
https://github.com/apache/mesos/blob/0.26.0-rc1/include/mesos/mesos.proto#L1383

I didn't realize they were going through a deprecation cycle -- last I looked it was a clean break. Let's be defensive, keep the code as-is.